modules/planning/lattice/trajectory_generation/backup_trajectory_generator.cc

```diff
@@ -55,9 +57,20 @@ void BackupTrajectoryGenerator::GenerateTrajectory1dPairs(
 
 DiscretizedTrajectory BackupTrajectoryGenerator::GenerateTrajectory(
     const std::vector<PathPoint>& discretized_ref_points) {
+  while (trajectory_pair_pqueue_.size() > 1) {
+    auto top_pair = trajectory_pair_pqueue_.top();
+    trajectory_pair_pqueue_.pop();
+    DiscretizedTrajectory trajectory = TrajectoryCombiner::Combine(
+        discretized_ref_points, *top_pair.first, *top_pair.second,
+        init_relative_time_);
+    if (!ptr_collision_checker_->InCollision(trajectory)) {
+      return trajectory;
+    }
+  }
   auto top_pair = trajectory_pair_pqueue_.top();
-  return TrajectoryCombiner::Combine(discretized_ref_points, *top_pair.first,
-                                     *top_pair.second, init_relative_time_);
+  return TrajectoryCombiner::Combine(
```

Comment:
The queue will be empty when reaches this line.

Reply:
Never mind.

Reply:
No. I keep the last one in the while loop.